crates/rpc/rpc/src/eth/api/server.rs

```diff
@@ -160,8 +162,85 @@ where
     }
 
     /// Handler for: `eth_getTransactionReceipt`
-    async fn transaction_receipt(&self, _hash: H256) -> Result<Option<TransactionReceipt>> {
-        Err(internal_rpc_err("unimplemented"))
+    async fn transaction_receipt(&self, hash: H256) -> Result<Option<TransactionReceipt>> {
+        trace!(target: "rpc::eth", ?hash, "Serving eth_getTransactionReceipt");
+
+        let receipt = match self.inner.client.receipt_by_hash(hash) {
+            Ok(Some(recpt)) => recpt,
+            Ok(None) => return Err(EthApiError::Unsupported("receipt not found").into()),
+            Err(_) => return Err(EthApiError::Unsupported("call resulted in error").into()),
+        };
+
+        let (tx, meta) = match self.client().transaction_by_hash_with_meta(hash) {
+            Ok(Some((tx, meta))) => (tx, meta),
+            Ok(None) => return Err(EthApiError::Unsupported("metadata not found").into()),
+            Err(_) => return Err(EthApiError::Unsupported("call resulted in error").into()),
+        };
+
+        let transaction =
+            tx.clone().into_ecrecovered().ok_or(EthApiError::InvalidTransactionSignature)?;
+
+        let mut res_receipt = TransactionReceipt {
+            transaction_hash: Some(meta.tx_hash),
+            transaction_index: Some(U256::from(meta.index)),
+            block_hash: Some(meta.block_hash),
+            block_number: Some(U256::from(meta.block_number)),
+            from: transaction.signer,
+            to: Some(Address::zero()),
+            cumulative_gas_used: U256::from(receipt.cumulative_gas_used),
+            gas_used: Some(U256::from(0)),
+            contract_address: None,
+            logs: receipt.logs.clone().into_iter().map(|l| l.into()).collect(),
+            effective_gas_price: U128::from(0),
+            transaction_type: U256::from(0),
+            state_root: None,
+            logs_bloom: receipt.bloom_slow(),
+            status_code: None,
+        };
+
+        if receipt.success {
+            res_receipt.status_code = Some(U64::from(1));
+        } else {
+            res_receipt.status_code = Some(U64::from(0));
+        }
+
+        match tx.transaction.kind() {
+            Create => {
+                res_receipt.to = Some(Address::zero());
+                // TODO get contract address
```

Comment:
@mattsse how to fetch the address of contract deployed?

Reply:
@rakita this isn't stored anywhere, right?
so we need to recompute via:

https://github.com/paradigmxyz/reth/blob/bd8802a8f8295e5dd93ce90e6bb7397f330d3e14/crates/primitives/src/contract.rs#L5-L5

Reply:
No I was asking for the `contract_address` field of the tx reciept.
https://github.com/paradigmxyz/reth/blob/bd8802a8f8295e5dd93ce90e6bb7397f330d3e14/crates/rpc/rpc-types/src/eth/transaction/receipt.rs#L25

Reply:
contract address can be calculated with nonce and caller address, `create_address` is the fn.
And it should be okay to calculate it only if `to` is `None` (Is create tx) and if `is_success` is `true` (contract was created).

Reply:
I see, only looked at the snippet, sorry.

this can be derived via:

https://github.com/paradigmxyz/reth/blob/bd8802a8f8295e5dd93ce90e6bb7397f330d3e14/crates/primitives/src/contract.rs#L5-L5

`create_address`

this is always correct, right @rakita ?